integration-tests/chopsticks/src/tests/switchPallet/fullFlowSwitch.test.ts

```diff
@@ -0,0 +1,177 @@
+import { test } from 'vitest'
+
+import * as PeregrineConfig from '../../network/peregrine.js'
+import * as AssetHubConfig from '../../network/assetHub.js'
+import {
+	KILT,
+	ROC,
+	getAssetSwitchParameters,
+	initialBalanceKILT,
+	initialBalanceROC,
+	keysAlice,
+	keysCharlie,
+} from '../../utils.js'
+import {
+	peregrineContext,
+	getFreeBalancePeregrine,
+	getFreeRocPeregrine,
+	getFreeEkiltAssetHub,
+	assethubContext,
+	checkSwitchPalletInvariant,
+} from '../index.js'
+import {
+	checkBalance,
+	createBlock,
+	setStorage,
+	hexAddress,
+	checkBalanceInRange,
+	getXcmMessageV4ToSendEkilt,
+} from '../utils.js'
+import { getAccountLocationV4, getRelayNativeAssetIdLocationV4, getSiblingLocationV4 } from '../../network/utils.js'
+import { sendTransaction, withExpect } from '@acala-network/chopsticks-testing'
+
+/**
+ * Full e2e flow between Peregrine and AssetHub. More checks are provided in individual test cases.
+ *
+ * 1. Send ROCs from AssetHub to Peregrine
+ * 2. Switch KILTs on Peregrine
+ * 3. Send eKILTs back to AssetHub
+ * 4. Send ROCs back to AssetHub (Currently not implemented)
+ */
+test('Full e2e tests', async ({ expect }) => {
```

Comment:
Yeah, prob not. It is not really possible yet. If I wrap those tests in different `describes` the `beforeAll` is not executed since they have a different context.

Reply:
Yeah ok ignore this. I won't resolve it so we can use it as a reference for the refactoring PR.